File: 2_python/Main_2884.py
hour,min = map(int,input().split())

hour=hour
min=min-45

if min>=0 : 
    print(hour,min)
else : 
    hour=hour-1
    min=min+60
    if hour>=0 :
        print(hour,min)
    else :
        hour = hour + 24
        print(hour,min)

File: 2_java/Main_2884.java
import java.util.Scanner;

public class Main_2884 {

    

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        int hour = input.nextInt();
        int min = input.nextInt();
        int totalMin = hour*60+min;

        int alarm = totalMin-45;

        if (alarm>=0) {
            System.out.println(alarm/60 + " " + alarm%60);
        } 
        else {
            System.out.println(((1440+alarm)/60)+" "+((1440+alarm)%60));
        }
        input.close();


    }
}
